python/fnllm/fnllm/services/cache_interactor.py

```diff
@@ -35,20 +52,22 @@ async def get_or_insert(
         name: str | None,
         json_model: type[TJsonModel],
         bypass_cache: bool = False,
-    ) -> TJsonModel:
+    ) -> Cached[TJsonModel]:
         """Get or insert an item into the cache."""
         if not self._cache or bypass_cache:
-            return await func()
+            result = await func()
+            return Cached(value=result, hit=False)
 
         key = self._cache.create_key(key_data, prefix=prefix)
-        cached_value = await self._cache.get(key)
-
-        if cached_value:
+        if self._cache.has(key):
```

Comment:
There might be some performance penalty in here (doing I/O at `.has` and `.get`), as well as room for race conditions. Instead, would be possible to call `.get` and check for a `None` result to see if it does not exist?